src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1527,6 +1528,53 @@ def reoffset(o):
 
         minimize_int(offset, reoffset)
 
+    def shrink_offset_pairs(self):
+        """Lower any two blocks offset from each other the same ammount.
+
+        Try lowering every non-zero pair of non-shrinking blocks by a
+        common offset.
+        """
+        self.debug('Shrinking offset pairs.')
+
+        def int_from_block(i):
+            u, v = self.blocks[i]
+            return int_from_bytes(self.shrink_target.buffer[u:v])
+
+        def block_len(i):
+            u, v = self.blocks[i]
+            return v - u
+
+        # Try reoffseting every pair
+        def reoffset_pair(pair, o):
+            n = len(self.blocks)
+            # Number of blocks may have changed, need to validate
+            valid_pair = [p for p in pair if p < n]
+
+            m = min([int_from_block(p) for p in valid_pair])
+
+            new_blocks = [self.shrink_target.buffer[u:v]
+                          for u, v in self.blocks]
+            for i in valid_pair:
+                if int_from_block(i) + o - m >= 0:
```

Comment:
The coverage job is complaining that this condition has never been false - you can just remove it.

This is because `int_from_bytes` interprets the bytes as an unsigned integer, and thus cannot be < 0.  That in turn is the source of all the terms, and `m` is *at most* equal to `o`.

If you wanted to, adding a simple docstring to the `int_(from|to)_bytes` implementations in compat.py (for both py2 and py3) would clarify this for future contributors: something like "Convert unsigned integer to big-endian bytes." and "Interpret bytes as a big-endian unsigned integer." would make sense to me.